pkg/scheduler/framework/plugins/helper/node_affinity.go

```diff
@@ -68,3 +68,33 @@ func nodeMatchesNodeSelector(node *v1.Node, nodeSelector *v1.NodeSelector) bool
 	matches, _ := corev1.MatchNodeSelectorTerms(node, nodeSelector)
 	return matches
 }
+
+// GetPodRequiredNodeSelectorAndAffinity returns required node selector and affinity for the pod.
+func GetPodRequiredNodeSelectorAndAffinity(pod *v1.Pod) (labels.Selector, *v1.NodeSelector) {
```

Comment:
Do not write new functions in this file.

Either reuse the existing functions in https://github.com/kubernetes/component-helpers/tree/master/scheduling/corev1/nodeaffinity or have them in the plugin

Reply:
+1 to move to `component-helpers` as it's used by more than one plugin.

Reply:
This was already done in #99959

Reply:
Can you rebase on top of it?

Reply:
Done.